### PR TITLE
Add wp crosstrack report for copter

### PR DESCRIPTION
**Pull request by khancyr**
It expose horizontal error of the actual position vs the desired position when navigating between waypoints and send it in nav_controller message

### CODE REVIEW

**Comment by rmackay9** [comment]
My only concern here is that it doesn't depend upon the flight mode.  This means that I think a cross track error will be reported even when the vehicle is in Stabilize mode.  A better approach may be to access this through the flight mode object so that it has a chance to provide the value only when we're doing straigh line waypoint navigation.

**Comment by khancyr** [comment]
@rmackay9 , I double check, and it is used only when we use the AC_WPNav so in guided, auto ,rtl, brake, circle.

I will  put that in the flight mode as I think it is clearer too

**Comment by lvale** [comment]
@khancyr Please add a functional description of the behaviour and results

File: ArduCopter/mode.h [hunk 1]
```diff
@@ -780,6 +782,7 @@ class ModeGuided : public Mode {
 
     uint32_t wp_distance() const override;
     int32_t wp_bearing() const override;
+    float crosstrack_error() const override { return wp_nav->crosstrack_error();}
```
Comment (OXINARF) [inline ArduCopter/mode.h]: This needs to depend on `guided_mode == Guided_WP`, other modes don't have a crosstrack error.

File: libraries/AC_WPNav/AC_WPNav.h [hunk 2]
```diff
@@ -283,6 +285,7 @@ class AC_WPNav
     Vector3f    _origin;                // starting point of trip to next waypoint in cm from ekf origin
     Vector3f    _destination;           // target destination in cm from ekf origin
     Vector3f    _pos_delta_unit;        // each axis's percentage of the total track from origin to destination
+    float       _track_error_xy;
```
Comment (OXINARF) [inline libraries/AC_WPNav/AC_WPNav.h]: Please add a comment like the others.

**Comment by Kiwa21** [comment]
@lvale It gives the horizontal error of the actual position vs the desired position when navigating between waypoints == it's the orthogonal projection of the real world position vs the ideal straight line between waypoints.

**Comment by lvale** [comment]
@Kiwa21 My comment is that the initial commit should contain the explanation

**Comment by khancyr** [comment]
Rebase and address comments from review

**Comment by rmackay9** [comment]
looking good, merged, thanks!

**Pull request merged**